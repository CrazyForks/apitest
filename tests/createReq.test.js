const { spwanTest } = require("./utils");

describe("createReq", () => {
  test("eval", async () => {
    const { stdout, code } = await spwanTest("createReq/eval.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("main", async () => {
    const { stdout, code } = await spwanTest("createReq", ["--ci"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
  test("mock", async () => {
    const { stdout, code } = await spwanTest("createReq/mock.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
});
