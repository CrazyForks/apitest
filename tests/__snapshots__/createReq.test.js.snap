// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createReq eval 1`] = `
"module main
  @eval complex expr ✔
  fail# @eval syntax error ✘
  fail# @eval must be string value ✘

1. fail# @eval syntax error(main.test2)
   main.test2.req.v1@eval: throw err, Unexpected identifier

2. fail# @eval must be string value(main.test3)
   main.test3.req.v1@eval: should have string value

"
`;

exports[`createReq main 1`] = `
"module main
  req variables ✔
  all kinds ✔
  nest value ✔

"
`;

exports[`createReq mock 1`] = `
"module main
  @mock ✔
  fail# @mock ✘
  fail# @mock must be string value ✘

1. fail# @mock(main.test2)
   main.test2.req.v1@mock: bad mock 'xyz'

2. fail# @mock must be string value(main.test3)
   main.test3.req.v1@mock: should have string value

"
`;
